src/components/sidenav/sidenav.js

```diff
@@ -323,6 +326,101 @@ function SidenavDirective($mdMedia, $mdUtil, $mdConstant, $mdTheming, $animate,
       }
     }
 
+    function enableDragging() {
+      $mdGesture.register(element, 'drag', { horizontal: true });
+
+      element
+        .on('$md.dragstart', onDragStart)
+        .on('$md.drag', onDrag)
+        .on('$md.dragend', onDragEnd);
+
+      var style = getComputedStyle(element[0]);
+      var sidenavWidth = parseInt(style.width);
+      var isRightSidenav = element.hasClass('md-sidenav-right');
+      var accelerationBound = 6;
+
+      var dragCancelled = false;
+      var dragPercentage;
+      var lastOpenState;
+      var lastDistance = 0;
+      var isQuickDrag = false;
+
+      function onDragStart() {
+        if (element.hasClass('md-locked-open')) {
+          dragCancelled = true;
+        } else {
+          lastOpenState = scope.isOpen;
+          element.css($mdConstant.CSS.TRANSITION_DURATION, '0ms');
+        }
+      }
+
+      function onDrag(ev) {
+        if (dragCancelled) return;
+
+        if (!isQuickDrag) {
+          var distance = lastDistance - ev.pointer.distanceX;
+          if (isRightSidenav) {
+            isQuickDrag = distance <= -accelerationBound;
+          } else {
+            isQuickDrag = distance >= accelerationBound;
+          }
+        } else {
+          if (isRightSidenav && lastDistance > ev.pointer.distanceX ||
+              !isRightSidenav && lastDistance < ev.pointer.distanceX) {
+            isQuickDrag = false;
+          }
+        }
+
+        dragPercentage = Math.round((ev.pointer.distanceX / sidenavWidth) * 100);
+        if (!isRightSidenav) dragPercentage = 0 - dragPercentage;
+
+        if (dragPercentage > 100) dragPercentage = 100;
+        else if (dragPercentage < 0) dragPercentage = 0;
+
+        element.css($mdConstant.CSS.TRANSFORM, 'translate3d(-' + (isRightSidenav ? 100 - dragPercentage : dragPercentage) + '%,0,0)');
+        lastDistance = ev.pointer.distanceX;
+      }
+
+      function onDragEnd() {
+        if (dragCancelled) {
+          dragCancelled = false;
+          return;
+        }
+
+        var remainingPercentage = 100 - dragPercentage;
+        var animationTime = 4 * remainingPercentage;
+        var isOpen = dragPercentage > 50 || isQuickDrag;
+        if (isRightSidenav) {
+          isOpen = !isOpen;
+        }
+
+        element.css($mdConstant.CSS.TRANSITION_DURATION, animationTime + "ms");
+
+        element.css($mdConstant.CSS.TRANSFORM, 'translate3d(' + (isOpen ? -100 : 0) +  '%,0,0)');
+
+        // Reset drag
+        lastDistance = 0;
+        isQuickDrag = false;
+        $timeout(onAnimationDone, animationTime, true, (isRightSidenav ? isOpen : !isOpen));
+      }
+
+      function onAnimationDone(isOpen) {
+        scope.isOpen = isOpen;
+        element.css($mdConstant.CSS.TRANSFORM, '');
+        element.css($mdConstant.CSS.TRANSITION_DURATION, '');
+
+        if (isOpen) {
+          if (!lastOpenState) {
+            $animate.enter(backdrop, element.parent());
+          }
+          element.removeClass('md-closed');
```

Comment:
That isn't working here, because we need to run the animation in the same digest.